chef/automate
# API to update the web ui key

### :nut_and_bolt: Description: What code changed, and why?


we need to add one API to update the webui key for the server.
### :chains: Related Resources
https://github.com/chef/automate/issues/6274
### :+1: Definition of Done
I have added API to update the webui key.
### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs
3. Add the web ui key in the request body

Request
```
curl --location --request POST 'https://a2-dev.test/api/v0/infra/servers/update' \
--header 'api-token: your admin token' \
--header 'Content-Type: text/plain' \
--data-raw '{
    "id":"local-dev",
    "webui_key": "---webui key----"
}'
```
Response
{}
### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
